app/assets/javascripts/discourse/app/widgets/post-menu.js

```diff
@@ -552,8 +552,18 @@ export default createWidget("post-menu", {
       postControls.push(repliesButton);
     }
 
-    let extraControls = applyDecorators(this, "extra-controls", attrs, state);
-    postControls.push(h("div.actions", visibleButtons.concat(extraControls)));
+    const extraControls = applyDecorators(this, "extra-controls", attrs, state);
+    const beforeExtraControls = applyDecorators(
+      this,
+      "before-extra-controls",
+      attrs,
+      state
+    );
+
+    let controlsButtons = beforeExtraControls.concat(visibleButtons);
+    controlsButtons = controlsButtons.concat(extraControls);
```

Comment:
Can we do this instead?

```suggestion
    const controlsButtons = [...beforeExtraControls, ...visibleButtons, ...extraControls];
```